packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts

```diff
@@ -123,6 +133,11 @@ const baseCases = [
     code: 'foo.bar && foo.bar?.() && foo.bar?.().baz',
     output: 'foo.bar?.()?.baz',
   },
+  // TODO: deepest left node already pre-optional chained
```

Comment:
What's the status of this? Is there pending work before we review the PR fully?

Reply:
I figured it's ok to add this comment for the case that did not work and was not covered as there is a similar comment a few lines above it: https://github.com/typescript-eslint/typescript-eslint/blob/befa01f0c8c9166748c491fad01d0d800a368bb9/packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts#L197-L198

Reply:
Same issue as with the `&&` base case, I opened an issue for